cli/azd/pkg/tools/azcli/spring_app.go

```diff
@@ -105,6 +147,108 @@ func (ss *springService) GetSpringAppProperties(
 	}, nil
 }
 
+func (ss *springService) GetSpringInstanceTier(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroup string,
+	instanceName string,
+) (*string, error) {
+	client, err := ss.createSpringClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Get(ctx, resourceGroup, instanceName, nil)
+	if err != nil {
+		return nil, err
+	}
+	return resp.SKU.Tier, nil
+}
+
+func (ss *springService) CreateBuild(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroupName string,
+	instanceName string,
+	buildServiceName string,
+	agentPoolName string,
+	builderName string,
+	buildName string,
+	jvmVersion string,
+	relativePath string,
+) (*string, error) {
+	client, err := ss.createBuildServiceClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	agentPoolId := "/subscriptions/" + subscriptionId +
+		"/resourceGroups/" + resourceGroupName +
+		"/providers/Microsoft.AppPlatform/Spring/" + instanceName +
+		"/buildServices/" + buildServiceName +
+		"/agentPools/" + agentPoolName
+	builderId := "/subscriptions/" + subscriptionId +
+		"/resourceGroups/" + resourceGroupName +
+		"/providers/Microsoft.AppPlatform/Spring/" + instanceName +
+		"/buildServices/" + buildServiceName +
+		"/builders/" + builderName
+
+	resp, err := client.CreateOrUpdateBuild(ctx, resourceGroupName, instanceName, buildServiceName, buildName,
+		armappplatform.Build{
+			Properties: &armappplatform.BuildProperties{
+				AgentPool: to.Ptr(agentPoolId),
+				Builder:   to.Ptr(builderId),
+				Env: map[string]*string{
+					"BP_JVM_VERSION": to.Ptr(jvmVersion),
+				},
+				RelativePath: to.Ptr(relativePath),
+			},
+		}, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Properties.TriggeredBuildResult.ID, nil
+}
+
+func (ss *springService) GetBuildResult(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroupName string,
+	instanceName string,
+	buildServiceName string,
+	buildName string,
+	buildResult string,
+) (*armappplatform.BuildResultProvisioningState, error) {
+	client, err := ss.createBuildServiceClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	buildResultName := buildResult[strings.LastIndex(buildResult, "/")+1:]
+	retries := 0
+	const maxRetries = 50
+	for {
+		resp, err := client.GetBuildResult(ctx, resourceGroupName, instanceName, buildServiceName, buildName, buildResultName, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		if *resp.Properties.ProvisioningState == armappplatform.BuildResultProvisioningStateSucceeded {
+			return resp.Properties.ProvisioningState, nil
+		} else if *resp.Properties.ProvisioningState == armappplatform.BuildResultProvisioningStateFailed {
+			return resp.Properties.ProvisioningState, errors.New("build result failed")
+		}
+		retries++
+
+		if retries >= maxRetries {
+			return nil, errors.New("get build result timeout")
+		}
+
+		time.Sleep(20 * time.Second)
+	}
```

Comment:
Consider using the `github.com/sethvargo/go-retry` library that we use in other parts of azd.  It would look something like:

```go
var result *armappplatform.BuildResultProvisioningState
err := retry.Do(ctx, retry.WithMaxRetries(50, retry.NewConstant(20*time.Second)), func(ctx context.Context) error {
              resp, err := client.GetBuildResult(ctx, resourceGroupName, instanceName, buildServiceName, buildName, buildResultName, nil)
		if err != nil {
                        // no retriable error
			return err
		}
    		
                if *resp.Properties.ProvisioningState == armappplatform.BuildResultProvisioningStateSucceeded {
			result = resp.Properties.ProvisioningState
                        return nil
		} else if *resp.Properties.ProvisioningState == armappplatform.BuildResultProvisioningStateFailed {
			result = resp.Properties.ProvisioningState
                        return errors.New("build result failed")
		}

                 // return error to retry
     	        return retry.RetryableError(fmt.Errorf("error but will retry: %w", err))
	})

return result, err
```

Reply:
Thanks, will refactor here soon